scipy/stats/_qmc.py

```diff
@@ -1036,6 +1084,56 @@ def _random(self, n: IntNumber = 1) -> np.ndarray:
         self.num_generated += n
         return samples
 
+    def _random_oa_lhs(self, n: IntNumber = 1) -> np.ndarray:
+        """Orthogonal array based LHS of strength 2."""
+        p = np.sqrt(n).astype(int)
+        n_row = p**2
+        n_col = p + 1
+
+        primes = primes_from_2_to(p + 2)
+        if p not in primes or n != n_row:
+            raise ValueError(
+                "n is not the square of a prime number. Closest"
+                f" values are {primes[-2:]**2}"
```

Comment:
It is very much an edge case but this logic does not necessarily produce "Closest values". When the input is `n = 11**2+ 1`, it suggests `7**2` and `11**2`, but the closest values should be `11**2` and `13**2`. Maybe you can just say "close values" instead of "closest values".

Reply:
Indeed, for this to work it would require to compute the next prime. I will update the warning, thanks!